docs/testing.md

```diff
@@ -0,0 +1,148 @@
+
+1. Delete the old version of docker (if it was installed)
+
+   ```bash
+   sudo dnf remove docker-*
+   sudo dnf config-manager --disable docker-*
+   ```
+
+2. As docker doesn't support **CGroups2**, enable old **CGroups**
+
+   ```bash
+   sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+   ```
+
+3. Configure firewall
+
+   ```bash
+   sudo firewall-cmd --permanent --zone=trusted --add-interface=docker0
+   sudo firewall-cmd --permanent --zone=FedoraWorkstation --add-masquerade
+   ```
+
+4. Install Moby (an open-source project created by Docker)
+
+   ```bash
+   sudo dnf install moby-engine docker-compose
+   sudo systemctl enable docker
+   ```
+
+5. Add your user to **docker** group to run docker without **sudo**
+
+   ```bash
+   sudo groupadd docker
+   sudo usermod -aG docker $USER
+   ```
+
+6. Restart your computer
+
+   ```bash
+   sudo reboot
+   ```
+
+7. To test your installation you need running the following command
+
+   ```bash
+   sudo docker run hello-world
+   ```
+
+#### Python configuration
+
+- Install `python3.6` package in system using your distributive package manager tool: **apt**, **dnf**, **yum**, etc.
+
+- Clone/Update the `cortx-prvsnr` repository
+
+- Create the Python virtual environment using `venv` module
+
+  ```bash
+  cd /path/to/cortx-prvsnr
+  python3.6 -m venv ./venv
+  ```
+
+  :warning: NOTE: For more details about Python virtual environment see
+  https://docs.python.org/3.6/library/venv.html
+
+- Install provisioner API to `venv`: 
+
+  ```bash
+  pip install -e api/python[test]
+  ```
+
+- Start the docker environment creation
+
+  ```bash
+  pytest test/build_helpers -k test_build_setup_env -s --root-passwd root --nodes-num 1
+  ```
+
+  The command above will create the docker container and will have the following output
+
+  ```
+  ========================================================================================= test session starts ==========================================================================================
+  platform linux -- Python 3.6.13, pytest-5.1.1, py-1.10.0, pluggy-0.13.1
+  rootdir: /home/fdmitry/Work/cortx-prvsnr, inifile: pytest.ini
+  plugins: timeout-1.3.4, testinfra-3.1.0, mock-3.1.0, xdist-1.29.0, forked-1.3.0
+  timeout: 7200.0s
+  timeout method: signal
+  timeout func_only: False
+  collected 1 item                                                                                                                                                                                       
+  
+  test/build_helpers/test_build_env.py::test_build_setup_env Host srvnode-1
+    Hostname 172.17.0.2
+    Port 22
+    User root
+    UserKnownHostsFile /dev/null
+    StrictHostKeyChecking no
+    IdentityFile /tmp/pytest-of-fdmitry/pytest-10/id_rsa.test
+    IdentitiesOnly yes
+    LogLevel FATAL
+  
+  ```
+
+- Install provisioner API to created docker container
```

Comment:
It's not only API. Better to say: You may then setup a provisioner inside the container using the `local` source mode.

And add marks:
TBD Test other types of sources.
TBD Update with examples for multi node setups with HA mode enabled.

Reply:
Done